### PR TITLE
Port tests/spec/benchmarks/encode-keccak00-spec.k to Haskell backend

### LINKED ISSUE #1011: Debug `make tests/specs/benchmarks/encode-keccak00-spec.k.prove TEST_SYMBOLIC_BACKEND=haskell`
1. `make ... --dry-run`
2. `kevm prove --backend haskell SPEC_FILE VERIFICATION_MODULE --concret .....`
3. `export PATH=$PATH:.build/usr/bin`

Then either do:
4. Add options `--debugger --debug-script kscript` <- if you want to use the repl/debugger.
5. `kevm prove .... &> output` <- will output pretty K

Example `kscript` file:

```
alias konfig = config | kast -i kore -o pretty -d .build/usr/lib/kevm/haskell /dev/stdin
```

Then in `kore-repl` you can call `konfig` to pretty-print a term. This _will not work_ because of the syntax declared inbetween the main module and the spec module in KEVM. So you'll need to move the definition of `#gas(...)` from `tests/specs/infinite-gas.k` to `edsl.md`, and change the Makefile for KEVM to use `EDSL` as the main module.

### PR DESCRIPTION
**Pull request by JKTKops**
~~I tentatively removed this test from `tests/failing-symbolic.haskell` as well.~~

fixes #1011

### CODE REVIEW

File: tests/specs/benchmarks/functional-spec.k [hunk 1]
```diff
@@ -17,4 +17,9 @@ module FUNCTIONAL-SPEC
 
     claim <k> runLemma(#encodeArgs(#bytes32(keccak(1 : #encodeArgs(#uint256(A0)))))) => doneLemma(#buf(32, keccak(#buf(1,1) ++ #buf(32, A0)))) ... </k> requires #rangeUInt(256, A0)
 
+    claim <k> runLemma(#buf(32, maxUInt256 &Int keccak(#buf(32, maxUInt256 &Int A0) ++ #buf(32, maxUInt256 &Int A1))))
+           => doneLemma(#buf(32, keccak(#buf(32,A0) ++ #buf(32,A1))))
+          </k>
```
Comment (ehildenb) [inline tests/specs/benchmarks/functional-spec.k:22]:
```suggestion
    claim <k> runLemma(#buf(32, maxUInt256 &Int keccak(#buf(32, maxUInt256 &Int A0) ++ #buf(32, maxUInt256 &Int A1))))
           => doneLemma(#buf(32, keccak(#buf(32,A0) ++ #buf(32,A1)))) ... </k>
```

**Review by ehildenb — APPROVED** [review]

**Comment by ehildenb** [comment]
Let's up the Jenkins timeout 20 minutes and try again with this test enabled @JKTKops .

**Review by Riib11 — APPROVED** [review]

**Review by ehildenb — CHANGES_REQUESTED** [review]
@JKTKops please remove the submodule bump to the K submodule present here. I guess you did a `git add -a`? But we should always let the auto-updater bump the submodule (see https://github.com/kframework/evm-semantics/pull/1019), because it (i) keeps the auto-updater from getting confused, and (ii) allows us to test submodule bumps in isolation and only include the changes needed to bump versions.

**Comment by JKTKops** [comment]
Removed the submodule bump - yes, I did `commit -am`.

**Review by ehildenb — APPROVED** [review]

**Review by Riib11 — APPROVED** [review]

**Pull request merged**